### PR TITLE
Integrator can pre-select a document type with or without country selection

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem
We want to allow integrators to be able to start the SDK with a preselected document type to decide their own document capture flow now that a country selection screen has been introduced for the `document` step.

# Solution
Added configuration option `skipCountrySelection` for `document` step. This option is disabled by default and only applies when only 1 document type is selected in `documentTypes` option.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [x] Has the README been updated?
- [n/a] Has the CONTRIBUTING doc been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [x] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [x] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [x] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Review by ethan605 — APPROVED** [review]
LGTM!

File: MIGRATION.md [hunk 1]
```diff
@@ -2,7 +2,77 @@
 
 The guides below are provided to ease the transition of existing applications using the Onfido SDK from one version to another that introduces breaking API changes.
 
-## `5.10.0` -> [next-version]
+## `5.10.0` -> `6.0.0`
+
+### Change in UX flow for Document step
+
+- Document step now has a Issuing Country Selection screen after the Document Type Selection screen. This screen is not displayed for **passport** documents. It is also skipped by default if the Document step is configured with driving licence or national ID card as the only preselected document type. This screen can still be included in the document capture flow by enabling the `showCountrySelection` option in the Document step configuration.
```
Comment (stefaniacardenas) [inline MIGRATION.md:9]: I found this description a bit confusing. How about

> [...]. This screen is never displayed for **passport** documents and is disabled by default when documents are preselected using the `documentTypes` option. This screen can still be included in the document capture flow of **non-passport** preselected documents by enabling the `showCountrySelection` option in the Document step configuration.

File: README.md [hunk 2]
```diff
@@ -411,19 +411,26 @@ A number of options are available to allow you to customise the SDK:
 
   This is the identity document capture step. Users will be asked to select the document type and to provide images of their selected document. For driving licence and national ID card types, the user will be able to see and select the supported countries. They will also have a chance to check the quality of the image(s) before confirming.
 
+  If you are migrating from a version older than `6.0.0` please refer to the [`MIGRATION`](MIGRATION.md) documentation for further details about this change.
```
Comment (stefaniacardenas) [inline README.md:414]: Why are we mentioning version `6.0.0` in the `README.md`? I think that integrators should refer to the migration file every time they upgrade to a major version. Also, in order to understand what "this change" means, an integrator would have to look at the commit history.

File: test/specs/scenarios/countrySelector.js [hunk 3]
```diff
@@ -57,6 +57,18 @@ export const countrySelectorScenarios = async (lang) => {
         verifyInitialUIElements(countrySelectorCopy)
       })
 
+      it("should skip country selection screen with a preselected driver's license document type", async () => {
```
Comment (stefaniacardenas) [inline test/specs/scenarios/countrySelector.js:60]: Would be good to add a test in the `poa` scenarios to make sure the screen isn't showing for poa docs either

File: src/demo/demoUtils.js [hunk 4]
```diff
@@ -7,6 +7,22 @@ export const queryParamToValueString = window.location.search
     return a
   }, {})
 
+const getDocumentTypes = () => {
```
Comment (stefaniacardenas) [inline src/demo/demoUtils.js:10]: I think it would be clearer to call this `getPreselectedDocumentTypes`

File: src/components/Router/StepComponentMap.js [hunk 5]
```diff
@@ -140,25 +143,50 @@ const getRequiredSelfieSteps = (deviceHasCameraSupport) => {
...
-  const withSelectScreen =
-    documentType === 'passport'
-      ? [SelectIdentityDocument, ...frontCaptureComponents]
-      : [SelectIdentityDocument, CountrySelector, ...frontCaptureComponents]
-  const frontDocumentFlow = hasPreselectedDocument
-    ? frontCaptureComponents
-    : withSelectScreen
-  if (double_sided_docs.includes(documentType)) {
+
+  if (isPassportDocument) {
```
Comment (stefaniacardenas) [inline src/components/Router/StepComponentMap.js:170]: The logic in this `getIdentityDocumentComponents` is getting complex. We should think of a way to refactor this (out of scope)

File: MIGRATION.md (hunk 1 above)
Comment (kopijunkie) [reply MIGRATION.md:9]: That sounds much better, thanks. Was struggling with how to document this. Though I think it's still important to be clear that it is only disabled by default if only 1 document is preselected.

File: README.md (hunk 2 above)
Comment (kopijunkie) [reply README.md:414]: Mentioned that version as it was the last major version. Was trying to be specific about when a major change was introduced. But that's a good point. I'll rephrase it to not refer to a previous version and be clearer what "this change" is.

File: src/components/Router/StepComponentMap.js (hunk 5 above)
Comment (kopijunkie) [reply src/components/Router/StepComponentMap.js:170]: Agreed. I tried to refactor it a bit here, but conscious of spending too much time on it.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1155-pr-onfido-sdk-ui-onfido.surge.sh

**Review by stefaniacardenas — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ This project adheres to the Node [default version scheme](https://docs.npmjs.com
 
 ### Added
 
-- UI: Add country selection step after document selection
+- UI: Add country selection screen after document selection. This screen is skipped by default for a preselected document but can still be displayed by enabling the `skipCountrySelection` option for the `document` step.
```
Comment (ethan605) [inline CHANGELOG.md:13]: It's a bit confusing here: should it be `showCountrySelection`?

**Review by ethan605 — CHANGES_REQUESTED** [review]
Just small typo

File: CHANGELOG.md (hunk 6 above)
Comment (kopijunkie) [reply CHANGELOG.md:13]: Yes, you're right. Good spot!

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Review by ethan605 — APPROVED** [review]
LGTM!

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1155-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1155-pr-onfido-sdk-ui-onfido.surge.sh
